From 4b3a5bd59bc7aa930b0afebecb485be27f768317 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Tue, 16 Jun 2020 20:48:43 +0000
Subject: [PATCH] Fix Qt 5.15 build

---
 .../videotexturebackend.pro                   |  3 +--
 1 file changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/videotexturebackend/videotexturebackend.pro b/src/videotexturebackend/videotexturebackend.pro
index 6f0cfdf..490db11 100644
--- a/src/videotexturebackend/videotexturebackend.pro
+++ b/src/videotexturebackend/videotexturebackend.pro
@@ -8,6 +8,7 @@ QT += \
         quick \
         multimedia \
         multimedia-private \
+        multimediagsttools-private \
         qtmultimediaquicktools-private
 
 CONFIG += plugin hide_symbols link_pkgconfig
@@ -17,8 +18,6 @@ PKGCONFIG +=\
         gstreamer-1.0 \
         nemo-gstreamer-interfaces-1.0
 
-LIBS += -lqgsttools_p
-
 # It won't compile without this,
 # the issue is Xlib.h defines Bool as int but  QJsonValue.h has an enum with Bool = 0x1 --> int = 0x1 -> BOOM!
 DEFINES += MESA_EGL_NO_X11_HEADERS
-- 
2.17.0

